src-tauri/src/parser/v1/mod.rs

```
@@ -237,6 +237,7 @@ pub struct PlayerState {
     pub index: u32,
     character_type: CharacterType,
     total_damage: u64,
+    last_known_pet_skill: Option<ActionType>,  // used for Ferry's skills that don't keep track of where they came from
```

Could also nix this and just throw unknown pet damage into the "Pets" bucket rather than trying to guess via this weird state management variable. Especially since it's not 100% accurate anyways (e.g. if you use BlausGespenst during Strafe there are two sources of pet skill damage happening at once and there's no simple way to figure out which is which).